### PR TITLE
Fix #988

### LINKED ISSUE #988: bug: unknown field for id
I ran into this when trying to index the id field of all my schemas, which was derived from a base mixin.

I am getting `entc/gen: invalid index for schema "User": unknown index field "id"` for that field, even though I know it exists and so does the database.

I tested, and this is present when the `id` field is not provided (default), when it is explicitly defined in the schema, and when it is provided by a mixin. Other fields work fine.

<details>
<summary><code>mixin.go</code></summary>

```go
// BaseMixin holds the schema definition for the BaseMixin entity.
type BaseMixin struct {
	mixin.Schema
}

// Fields of the Mixin.
func (BaseMixin) Fields() []ent.Field {
	return []ent.Field{
		field.UUID("id", uuid.UUID{}).Default(uuid.New),
		field.String("someOtherField"),
	}
}

```

OR

```go
// BaseMixin holds the schema definition for the BaseMixin entity.
type BaseMixin struct {
	mixin.Schema
}

// Fields of the Mixin.
func (BaseMixin) Fields() []ent.Field {
	return []ent.Field{
		field.UUID("id", uuid.UUID{}).Default(uuid.New),
		field.String("someOtherField"),
	}
}

func (BaseMixin) Indexes() []ent.Index {
	return []ent.Index{
		index.Fields("id", "someOtherField"),
	}
}
```
</details>

<details>
<summary><code>user.go</code></summary>

```go
// User holds the schema definition for the User entity.
type User struct {
	ent.Schema
}

// Fields of the User.
func (User) Fields() []ent.Field {
	return []ent.Field{
		field.String("testField"),
	}
}

// Edges of the User.
func (User) Edges() []ent.Edge {
	return nil
}

func (User) Indexes() []ent.Index {
	return []ent.Index{
		index.Fields("id"),
	}
}
```

OR

```go
// User holds the schema definition for the User entity.
type User struct {
	ent.Schema
}

func (User) Mixin() []ent.Mixin {
	return []ent.Mixin{
		BaseMixin{},
	}
}

// Fields of the User.
func (User) Fields() []ent.Field {
	return []ent.Field{
		field.String("testField"),
	}
}

// Edges of the User.
func (User) Edges() []ent.Edge {
	return nil
}

func (User) Indexes() []ent.Index {
	return []ent.Index{
		index.Fields("id"),
	}
}
```
</details>

### PR DESCRIPTION
**Pull request by cliedeman**
Fixes #988

### CODE REVIEW

File: entc/gen/type.go [hunk 1]
```diff
@@ -483,9 +483,16 @@ func (t *Type) AddIndex(idx *load.Index) error {
 		return fmt.Errorf("missing fields or edges")
 	}
 	for _, name := range idx.Fields {
-		f, ok := t.fields[name]
-		if !ok {
-			return fmt.Errorf("unknown index field %q", name)
+		var f *Field
+
```
Comment (a8m) [inline entc/gen/type.go:487]:
```suggestion
		var f *Field
```

File: entc/integration/customid/ent/schema/mixinid.go [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+// BaseMixin holds the schema definition for the BaseMixin entity.
+type BaseMixin struct {
+	mixin.Schema
+}
+
+// Fields of the Mixin.
+func (BaseMixin) Fields() []ent.Field {
+	return []ent.Field{
+		field.UUID("id", uuid.UUID{}).Default(uuid.New),
+		field.String("someOtherField"),
```
Comment (a8m) [inline entc/integration/customid/ent/schema/mixinid.go:24]: My convention is to use "snake_case" for columns field names.

File: entc/integration/customid/ent/schema/mixinid.go [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+
+// MixinID holds the schema definition for the MixinID entity.
+type MixinID struct {
+	ent.Schema
+}
+
+// Fields of the MixinID.
+func (MixinID) Fields() []ent.Field {
+	return []ent.Field{
+		field.String("testField"),
```
Comment (a8m) [inline entc/integration/customid/ent/schema/mixinid.go:36]: ditto

File: entc/integration/customid/ent/schema/mixinid.go [hunk 4]
```diff
@@ -0,0 +1,55 @@
...
+	}
+}
+
+// Edges of the MixinID.
+func (MixinID) Edges() []ent.Edge {
+	return nil
+}
```
Comment (a8m) [inline entc/integration/customid/ent/schema/mixinid.go:43]:
```suggestion
```

File: entc/integration/customid/ent/schema/mixinid.go [hunk 5]
```diff
@@ -0,0 +1,55 @@
...
+
+func (MixinID) Indexes() []ent.Index {
+	return []ent.Index{
+		index.Fields("id"),
+	}
+}
```
Comment (a8m) [inline entc/integration/customid/ent/schema/mixinid.go:49]: Add please a composite index with the `id` column.

**Review by a8m — COMMENTED** [review]
Looks great. I've added a few minor comments

Thanks a lot for your contribution!

**Comment by cliedeman** [comment]
@a8m comments applied

**Pull request closed without merging**